evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -2796,3 +2796,76 @@ def test_training_only_component_in_pipeline_transform(X_y_binary):
     pipeline.fit(X, y)
     transformed = pipeline.transform(X)
     assert len(transformed) == len(X) - 2
+
+
+def test_component_graph_pipeline():
+    component_graph1 = ComponentGraph(
+        {
+            "Imputer": ["Imputer", "X", "y"],
+            "Undersampler": ["Undersampler", "Imputer.x", "y"],
+            "Logistic Regression Classifier": [
+                "Logistic Regression Classifier",
+                "Undersampler.x",
+                "Undersampler.y",
+            ],
+        }
+    )
+
+    component_graph2 = ComponentGraph(
+        {
+            "Imputer": ["Imputer", "X", "y"],
+            "Linear Regressor": [
+                "Linear Regressor",
+                "Imputer.x",
+                "y",
+            ],
+        }
+    )
+
+    component_graph3 = ComponentGraph(
+        {
+            "Imputer": ["Imputer", "X", "y"],
+            "Undersampler": ["Undersampler", "Imputer.x", "y"],
+        }
+    )
+
+    BinaryClassificationPipeline(component_graph1)
```

Comment:
Can we add `BinaryClassificationPipeline(component_graph1).component_graph == componentgraph1` to make sure everything is wired correctly under the hood?